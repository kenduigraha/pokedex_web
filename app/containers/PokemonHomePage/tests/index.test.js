/**
 *
 * Tests for PokemonHomePage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import configureStore from '../../../configureStore';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { PokemonHomePage } from '../index';
// import { DEFAULT_LOCALE } from '../../../i18n';

describe('<PokemonHomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            <PokemonHomePage />
          </IntlProvider>
        </Provider>,
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
