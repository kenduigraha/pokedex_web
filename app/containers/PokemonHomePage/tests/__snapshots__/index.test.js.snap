// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonHomePage /> should render and match the snapshot 1`] = `
<div>
  <div>
    <div
      className="ant-row PokemonFilter__PokemonFilterRow-sc-1gevdxe-1 WQhsS"
      span={24}
      style={
        Object {
          "marginLeft": -10,
          "marginRight": -10,
        }
      }
    >
      <div
        className="ant-col ant-col-10"
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <div
          className="ant-select-lg PokemonFilter__PokemonSelect-sc-1gevdxe-0 fugVwZ ant-select ant-select-enabled"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            aria-autocomplete="list"
            aria-controls="test-uuid"
            aria-expanded={false}
            aria-haspopup="true"
            className="ant-select-selection
            ant-select-selection--single"
            onKeyDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            <div
              className="ant-select-selection__rendered"
            >
              <div
                className="ant-select-selection__placeholder"
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "display": "block",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                Select Pokémon's Type
              </div>
            </div>
            <span
              className="ant-select-arrow"
              onClick={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <i
                aria-label="icon: down"
                className="anticon anticon-down ant-select-arrow-icon"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </div>
        </div>
      </div>
      <div
        className="ant-col ant-col-14"
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <span
          className="ant-input-search ant-input-affix-wrapper ant-input-affix-wrapper-lg"
        >
          <input
            className="ant-input ant-input-lg"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search for Pokémon"
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-suffix"
          >
            <i
              aria-label="icon: search"
              className="anticon anticon-search ant-input-search-icon"
              onClick={[Function]}
              tabIndex={-1}
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </i>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="styles__PokemonListDiv-sc-1b5lzlw-0 cJsYwM"
    id="content"
  >
    <div>
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      />
    </div>
  </div>
</div>
`;
